### PR TITLE
Let cask doctor exit with 1

### PR DESCRIPTION
**Pull request by amyspark**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
(I am reopening #3418 with this pull request, since it got stale)

This pull request fixes caskroom/homebrew-cask#40198 by raising a CaskError if one or more of the checks fails.

Please review!

### CODE REVIEW

File: Library/Homebrew/cask/lib/hbc/cli/doctor.rb [hunk 1]
```diff
@@ -1,24 +1,114 @@
...
       def run
+        check_software_versions
+        check_install_location
+        check_staging_location
+        check_taps
+        check_load_path
+        check_environment_variables
+
+        puts summary unless success?
+        raise CaskError, "Your system is not ready for Cask." unless success?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/cli/doctor.rb]: Maybe “There are some problems with your setup.”.

File: Library/Homebrew/cask/lib/hbc/cli/doctor.rb [hunk 2]
```diff
@@ -1,24 +1,114 @@
...
         return if args.empty?
         raise ArgumentError, "#{self.class.command_name} does not take arguments."
       end
 
+      def success?
+        !(errors? || warnings?)
+      end
+
+      def summary_header
+        "Cask's Doctor checkup"
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/cli/doctor.rb]: Capitalize “checkup”.

**Comment by amyspark** [comment]
@reitermarkus, can you check if this is all right?

**Comment by MikeMcQuaid** [comment]
Thanks @amyspark!

**Comment by reitermarkus** [comment]
Thanks, @amyspark! I was about to get back to this today, but looks like Mike beat me by a day. 😉

**Pull request merged**